runtime/compiler/x/runtime/X86LockReservation.asm

```diff
@@ -109,7 +109,23 @@ endm
 TryLock macro
     push _rbp
     lea  _rbp, [_rbp + eq_J9Monitor_RESINCBits] ; make thread ID + RES + INC_DEC value
-    xor  _rax, _rax                             ; make eax 0 for the compare and exchange
+
+    ; Set _rax to the lock word value that allows a monitor to be reserved (the
+    ; reservation bit by default, or 0 for -XlockReservation). This value is
+    ; provided to the reserving monent helper as the third argument. Calling
+    ; conventions differ...
+    ifdef TR_HOST_32BIT
+        ; 32-bit always uses fastcall. This is the only argument on the stack.
+        ; Offset is +4 for saved ebp, and +4 for the return address.
+        mov eax, dword ptr [esp+8]
```

Comment:
Did you mean +0 for saved ebp?

Reply:
The saved value of ebp is at offset +0, but what I was trying to say is that it contributes +4 to the offset of the argument. In my first attempt at loading this argument, I had forgotten that ebp is also on the stack here.